packages/volto/src/helpers/Robots/Robots.js

```diff
@@ -15,41 +17,23 @@ import { addHeadersFactory } from '@plone/volto/helpers/Proxy/Proxy';
  */
 export const generateRobots = (req) =>
   new Promise((resolve) => {
-    const internalUrl =
-      config.settings.internalApiPath ?? config.settings.apiPath;
-    const request = superagent.get(`${internalUrl}/robots.txt`);
-    request.set('Accept', 'text/plain');
+    const request = superagent.get(formatUrl('@site'));
+    request.set('Accept', 'application/json');
     const authToken = req.universalCookies.get('auth_token');
     if (authToken) {
       request.set('Authorization', `Bearer ${authToken}`);
     }
     request.use(addHeadersFactory(req));
-    request.end((error, { text }) => {
+    request.end((error, { text, body }) => {
       if (error) {
         resolve(text || error);
       } else {
-        // It appears that express does not take the x-forwarded headers into
```

Comment:
@robgietema This was added by @reebalazs at some point, can't remember why, maybe it was relevant for something. Could you please ask him?